File: tests/official-sdk-qualification/python-bolt/qualification.py
import os

from slack_bolt import App
from slack_bolt.authorization import AuthorizeResult
from slack_bolt.request import BoltRequest
from slack_sdk import WebClient


token = os.environ.get("SAMEOLDCHAT_API_TOKEN", "xoxb-test")
base_url = os.environ.get("SAMEOLDCHAT_API_URL", "http://127.0.0.1:18080/api/")


def authorize(**_kwargs):
    return AuthorizeResult(
        enterprise_id=None,
        team_id="T1",
        bot_id="B1",
        bot_user_id="U1",
        bot_token=token,
    )


app = App(
    client=WebClient(token=token, base_url=base_url),
    authorize=authorize,
    signing_secret="qualification-only",
)

received = False


@app.event("message")
def handle_message(event, client):
    global received
    received = True
    assert event["channel"] == "C1"
    assert event["text"] == "qualification event"
    assert client.api_test()["ok"] is True


response = app.dispatch(
    BoltRequest(
        body={
            "type": "event_callback",
            "team_id": "T1",
            "api_app_id": "A1",
            "event_id": "Ev1",
            "event_time": 1,
            "event": {
                "type": "message",
                "channel": "C1",
                "user": "U2",
                "text": "qualification event",
                "ts": "1.000000",
                "event_ts": "1.000000",
            },
        },
        mode="socket_mode",
    )
)
assert response.status == 200
assert received is True, response.body
print("python-bolt qualification passed")

File: tests/official-sdk-qualification/node-bolt/qualification.mjs
import assert from "node:assert/strict";
import { App } from "@slack/bolt";

const apiUrl = process.env.SAMEOLDCHAT_API_URL ?? "http://127.0.0.1:18080/api/";
const token = process.env.SAMEOLDCHAT_API_TOKEN ?? "xoxb-test";
const app = new App({
  signingSecret: "qualification-signing",
  clientOptions: { slackApiUrl: apiUrl },
  authorize: async () => ({ botToken: token, teamId: "T1", userId: "U1" }),
});

let resolveReceived;
let rejectReceived;
const received = new Promise((resolve, reject) => {
  resolveReceived = resolve;
  rejectReceived = reject;
});
app.event("reaction_added", async ({ event, client }) => {
  try {
    assert.equal(event.item.channel, "C1");
    assert.equal(event.reaction, "wave");
    assert.equal(event.user, "U1");
    assert.equal((await client.api.test()).ok, true);
    resolveReceived();
  } catch (error) {
    rejectReceived(error);
  }
});

await app.start(19090);
try {
  const response = await fetch("http://127.0.0.1:18080/qualification/bolt-event", { method: "POST" });
  assert.equal(response.status, 204, await response.text());
  await Promise.race([
    received,
    new Promise((_, reject) => setTimeout(() => reject(new Error("Bolt did not receive the signed Events API request")), 3000)),
  ]);
} finally {
  await app.stop();
}
console.log("node-bolt qualification passed");
